app/models/manageiq/providers/amazon/network_manager/refresh_parser.rb

```diff
@@ -167,6 +219,109 @@ def parse_firewall_rule(perm, direction)
     ret
   end
 
+  def parse_load_balancer(lb)
+    uid = lb.load_balancer_name
+
+    new_result = {
+      :type    => self.class.load_balancer_type,
+      :ems_ref => uid,
+      :name    => uid,
+    }
+
+    return uid, new_result
+  end
+
+  def parse_load_balancer_pool(lb)
+    uid  = lb.load_balancer_name
+    name ||= uid
+
+    new_result = {
+      :type          => self.class.load_balancer_pool_type,
+      :ems_ref       => uid,
+      :name          => name,
+    }
+
+    return uid, new_result
+  end
+
+  def parse_load_balancer_pool_member(member)
+    uid = member.instance_id
+
+    {
+      :type    => self.class.load_balancer_pool_member_type,
+      :ems_ref => uid,
+      # TODO(lsmola) AWS always associates to eth0 of the instances, we do not collect that info now, we need to do that
+      # :network_port => get eth0 network_port
+      :vm      => parent_manager_fetch_path(:vms, uid)
+    }
+  end
+
+  def parse_load_balancer_listener(lb, listener_struct)
+    listener = listener_struct.listener
+
+    uid = "#{lb.load_balancer_name}__#{listener.protocol}__#{listener.load_balancer_port}__"\
+          "#{listener.instance_protocol}__#{listener.instance_port}__#{listener.ssl_certificate_id}"
+
+    new_result = {
+      :type                         => self.class.load_balancer_listener_type,
+      :ems_ref                      => uid,
+      :load_balancer_protocol       => listener.protocol,
+      :load_balancer_port           => listener.load_balancer_port,
+      :instance_protocol            => listener.instance_protocol,
+      :instance_port                => listener.instance_port,
+      :load_balancer                => @data_index.fetch_path(:load_balancers, lb.load_balancer_name),
+      :load_balancer_listener_pools => [
+        {:load_balancer_pool => @data_index.fetch_path(:load_balancer_pools, lb.load_balancer_name)}]
+    }
+
+    return uid, new_result
+  end
+
+  def parse_load_balancer_health_check(lb)
+    uid = lb.load_balancer_name
+
+    health_check_members = @aws_elb.client.describe_instance_health(:load_balancer_name => lb.load_balancer_name)
+    health_check_members = health_check_members.instance_states.collect do |m|
+      parse_load_balancer_health_check_member(m)
+    end
+
+    health_check = lb.health_check
+    target_match = health_check.target.match(/^(\w+)\:(\d+)\/?(.*?)$/)
+    protocol     = target_match[1]
+    port         = target_match[2].to_i
+    url_path     = target_match[3]
+
+    matched_listener = @data.fetch_path(:load_balancer_listeners).detect do |x|
```

Comment:
x,y and z are my best friends :-) it was inline before, which I changed because of too long line